bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -0,0 +1,243 @@
+        this.bindingConf = configuration;
+        this.httpService = httpService;
+        this.connectApi = new AuthenticationApi(this, scheduler);
+        this.httpClient = httpClient;
+        this.deserializer = deserializer;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Netatmo API bridge handler.");
+        thingConf = getConfigAs(ApiHandlerConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            openConnection();
+            String webHookUrl = thingConf.webHookUrl;
+            if (webHookUrl != null && !webHookUrl.isBlank()) {
+                servlet = Optional.of(new NetatmoServlet(httpService, this, webHookUrl));
+            }
+        });
+    }
+
+    private void openConnection() {
+        try {
+            Credentials credentials = thingConf.getCredentials();
+            logger.debug("Connecting to Netatmo API.");
+            try {
+                connectApi.authenticate(credentials, bindingConf.features);
+                updateStatus(ThingStatus.ONLINE);
+                getThing().getThings().forEach(child -> {
+                    if (child instanceof CommonInterface) {
+                        ((CommonInterface) child).expireData();
+                    }
+                });
+            } catch (NetatmoException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                prepareReconnection();
+            }
+        } catch (NetatmoException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void prepareReconnection() {
+        connectApi.disconnect();
+        freeConnectJob();
+        connectJob = Optional
+                .of(scheduler.schedule(() -> openConnection(), thingConf.reconnectInterval, TimeUnit.SECONDS));
+    }
+
+    private void freeConnectJob() {
+        connectJob.ifPresent(j -> j.cancel(true));
+        connectJob = Optional.empty();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Shutting down Netatmo API bridge handler.");
+        servlet.ifPresent(servlet -> servlet.dispose());
+        servlet = Optional.empty();
+        connectApi.dispose();
+        freeConnectJob();
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Netatmo Bridge is read-only and does not handle commands");
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(NetatmoDiscoveryService.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends RestManager> @Nullable T getRestManager(Class<T> clazz) {
+        if (!managers.containsKey(clazz)) {
+            try {
+                Constructor<T> constructor = clazz.getConstructor(getClass());
```

Comment:
This is apparently working but I don't understand why.
You search a constructor having a ApiBridgeHandler as parameter ?
But AircareApi or WeatherApi for example do not have this constructor.
What do I miss ?

Reply:
AircareApi and WeatherApi does have constructors with ApiBridgeHandler as parameter : 

```
    public WeatherApi(ApiBridgeHandler apiClient) {
        super(apiClient, FeatureArea.WEATHER);
    }

```

```
    public AircareApi(ApiBridgeHandler apiClient) {
        super(apiClient, FeatureArea.AIR_CARE);
    }
```

Reply:
Oh yes, certainly ;)
I was searching in capability classes rather than api classes !